src/main/java/lotto/domain/LottoNumbers.java

```diff
@@ -0,0 +1,59 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class LottoNumbers {
+    private final List<LottoNumber> numbers;
+
+    public LottoNumbers(List<Integer> numbers) {
+        this.numbers = toLottoNumber(numbers);
+    }
+
+    private static List<LottoNumber> toLottoNumber(List<Integer> numbers) {
+        List<LottoNumber> lottoNumbers = new ArrayList<>();
+
+        for (int number : numbers) {
+            lottoNumbers.add(new LottoNumber(number));
+        }
+
+        return lottoNumbers;
+    }
+
+    public LottoRank getMatchedRank(LottoNumbers other) {
+        int matchCount = 0;
+
+        for (int i = 0; i < numbers.size(); i++) {
+            if (matchesAtIndex(i, other)) {
+                matchCount++;
+            }
+        }
```

Comment:
```suggestion
        for (LottoNumber number: numbers) {
            if (other.contains(number)) {
                matchCount++;
            }
        }
```
로또는 굳이 위치 값까지 같지 않아도 됨
값이 있는지 여부를 판단하면 됨